utils_test.py

```diff
@@ -610,20 +690,27 @@ def test_require_valid_thumbnail_filename(self):
         utils.require_valid_thumbnail_filename(filename)
 
     def test_get_time_in_millisecs(self):
+        # type: () -> None
         dt = datetime.datetime(2020, 6, 15)
         msecs = utils.get_time_in_millisecs(dt)
         self.assertEqual(
             dt,
-            datetime.datetime.fromtimestamp(python_utils.divide(msecs, 1000.0)))
+            datetime.datetime.fromtimestamp(
+                python_utils.divide( # type: ignore[no-untyped-call]
+                    msecs, 1000.0)))
 
     def test_get_time_in_millisecs_with_complicated_time(self):
+        # type: () -> None
```

Comment:
Is there any way to disable the type for test functions? (I don't think this is useful and it's never going to change, right?)

Reply:
We decided to type even tests since it is not that hard.

Reply:
But it adds lots of extra types testing + unwanted information. (Note: We have also excluded having docstring for test_* functions)

Reply:
And other point was that we had a look at other organisations and found that they also add type annotations to test files. Example: Zulip.

Reply:
> And other point was that we had a look at other organisations and found that they also add type annotations to test files. Example: Zulip.

Thanks for the info!

I'm fine with leaving it as is if it's not possible to exclude test_* function for typecheck!

Reply:
We are already excluding a lot of files and excluding test files will not be a problem. I, @sajalasati , @vojtechjelinek  and @seanlip had a long discussion about this and the main reason for us to add type annotations to test files was that other organisations also add them. Thanks!

Reply:
> Is there any way to disable the type for test functions? (I don't think this is useful and it's never going to change, right?)

@DubeySandeep - @hardikkat24 told me in an offline discussion that we can only exclude complete files and not test methods. So what do you think we can exclude the entire test files? Else we will have to proceed this way only I guess - right @hardikkat24?

Reply:
Yes. Files can be ignored. Not methods.